pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -132,3 +133,43 @@ def __init__(self, set_variable):
         self._precision = set_variable._precision
         self._class_type = set_variable._class_type
         super().__init__(set_variable)
+
+
+class SetRemove(SetMethod):
+    """
+    Represents a call to the .remove() method.
+
+    The remove() method removes the specified item from 
+    the set and updates the set. It doesn't return any value.
+        The name of the set.
+
+    Parameters
+    ----------
+    set_variable : TypedAstNode
+        The set on which the method will operate.
+
+    item : TypedAstNode
+        The item to search for, and remove.
+    """
+    __slots__ = ()
+    _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = None
+    _class_type = NativeVoid()
+    name = 'remove'
+
+    def __init__(self, set_variable, item) -> None:
+        if not isinstance(item, TypedAstNode):
+            raise TypeError(f"KeyError : It is not possible to look for a {item} in a set of {set_variable.dtype}")
```

Comment:
I am not sure I agree with this error message. Can you give me an example of what it prints exactly?

Reply:
Here's an example : 

```python
     s = {1,2,3,4}
     s.remove(3.2)
```
The error message will be: `Can't remove an element of type float from a set of int (s.remove(3.2))`

Reply:
@mustapha-belbiad your comment is about the error message 10 lines below. Yaman is asking about the error message at line 164

Reply:
Sorry, Here's the example : 
```python
 s = {2, 3, 4, 5, 6}
 s.remove(range(3,5))
```
The error message will be: `It is not possible to look for a <pyccel.ast.builtins.PythonRange object at 0x7f957ac96920> in a set of int (s.remove(range(3, 5)))`

Reply:
![image](https://github.com/pyccel/pyccel/assets/25363473/fc2eba6d-c8fe-4ad5-8095-278e060ee5e3)

The error message is slightly longer than you describe

Reply:
I don't understand why it contains the term "KeyError"

Reply:
> I don't understand why it contains the term "KeyError"

In this case, Python displays this error term.

Reply:
Fixed : f3d2e96